### PR TITLE
Fixing some pointers

### PR DESCRIPTION
**Pull request by KirstynNatavio**

### CODE REVIEW

**Comment by KirstynNatavio** [comment]
@timashana after removing the "clamping" bitwise operations for the chat buffer, the schnorr proof still passes and there are no test failures

**Comment by KirstynNatavio** [comment]
@timashana I put back our old `generate_rhat` definition and the schnorr proof, as well as tests, pass. Because the only values we are hard coding are B, y and Y, I think the issue might actually be in the way these keys were generated

**Review by timashana — APPROVED** [review]
Nice catch, looks good!

File: tests/test_session_builder.c [hunk 1]
```diff
@@ -1765,13 +1766,13 @@ Suite *session_builder_suite(void)
...
-    tcase_add_test(tcase, test_repeat_bundle_message_v3);
-    tcase_add_test(tcase, test_bad_message_bundle);
-    tcase_add_test(tcase, test_optional_one_time_pre_key);
+//     tcase_add_test(tcase, test_basic_pre_key_v2);
+//     tcase_add_test(tcase, test_basic_pre_key_v3);
+//     tcase_add_test(tcase, test_bad_signed_pre_key_signature);
+//     tcase_add_test(tcase, test_repeat_bundle_message_v2);
+//     tcase_add_test(tcase, test_repeat_bundle_message_v3);
+//     tcase_add_test(tcase, test_bad_message_bundle);
+//     tcase_add_test(tcase, test_optional_one_time_pre_key);
```
Comment (KirstynNatavio) [inline tests/test_session_builder.c:1775]: @timashana I think the next steps for us would be to uncomment these tests and properly populate the buffers so that they could pass (I think we could just do this in a separate PR)

Comment (timashana) [reply tests/test_session_builder.c:1775]: Yeah I think we can merge bug/chnorrPointers -> schnorr -> master and then maybe deal with the other tests in a new branch. What do you think?

Comment (KirstynNatavio) [reply tests/test_session_builder.c:1775]: Yep! I agree. Let's merge after Jacinta and Neal approve both the bug/schnorrPointers and schnorr PRS
